src/v/bytes/random.cc

```diff
@@ -37,4 +38,25 @@ iobuf make_iobuf(size_t n) {
     return io;
 }
 
+iobuf split_as_iobuf(std::string_view str, int n_fragments) {
+    auto remaining_len = str.length();
+    std::vector<size_t> fragment_lengths;
+    fragment_lengths.reserve(n_fragments);
+    for (int i = 0; i < n_fragments - 1; ++i) {
+        size_t fragment_len = get_int(remaining_len);
+        fragment_lengths.push_back(fragment_len);
+        remaining_len -= fragment_len;
+    }
+    fragment_lengths.push_back(remaining_len);
+    std::ranges::shuffle(fragment_lengths, internal::gen);
+
+    iobuf res;
+    auto cur = str.begin();
+    for (auto fragment_len : fragment_lengths) {
+        res.append(ss::temporary_buffer<char>(cur, fragment_len));
+        cur += fragment_len;
+    }
+    return res;
```

Comment:
this should probably go into a random-specific test utility/library. i can't imagine this being useful in production right?

Reply:
By the function right above, which is also only used in tests, I judged that this library is a good place for this kind of function. Do you think we should move both to, say, `//src/v/random:generators`? It's used in production code too BTW, we don't seem to have a test-only library for randomization.

Reply:
> It's used in production code too BTW

where is it used in production? i only see these non-production usages

```
nwatkins@acid:~/src/redpanda$ git grep split_as_iobuf
src/v/bytes/random.cc:iobuf split_as_iobuf(std::string_view str, int n_fragments) {
src/v/bytes/random.h:iobuf split_as_iobuf(std::string_view str, int n_fragments);
src/v/hashing/tests/hash_bench.cc:    iobuf split_buf = random_generators::split_as_iobuf(
src/v/iceberg/tests/transform_utils_test.cc:        auto split = random_generators::split_as_iobuf(
src/v/iceberg/tests/transform_utils_test.cc:        auto split = random_generators::split_as_iobuf(
```

why would we need to randomly split a iobuf in a non-test situation?

Reply:
I meant `/src/v/random:generators` is used in production.

Reply:
right, so you can move the random splitting utility into a test-only library

Reply:
as an example, we have `test_utils/randoms.h` -- not that this should go there, i'm just highlighting that there is precedence.

Reply:
> not that this should go there

Why not? Especially if we combine it with data generation and call it `random_iobuf`.

Reply:
i'm not saying it shouldn't go there, i'm just saying it's an example of a place it could go.